src/Handler/Info/ObjectInfoHandler.php

```diff
@@ -0,0 +1,200 @@
+    {
+        assert(count($path) === 1, 'Error path must have exactly one element.');
+        $objects = new RepeatedField(GPBType::MESSAGE, ObjectInfo::class);
+        foreach ($this->getChildObjectsForSchema($db, $path[0]) as $object) {
+            $objects[] = $object;
+        }
+        $this->assertDatabaseExists($objects, $db, $path[0]);
+
+        $infoObject = new DatabaseInfo();
+        $infoObject->setObjects($objects);
+
+        $response->setDatabaseInfo($infoObject);
+
+        return $response;
+    }
+
+    /**
+     * @param string[] $path
+     */
+    private function getSchemaResponse(array $path, Connection $db, ObjectInfoResponse $response): ObjectInfoResponse
+    {
+        assert(count($path) === 1, 'Error path must have exactly one element.');
+        $infoObject = new SchemaInfo();
+        $objects = new RepeatedField(GPBType::MESSAGE, ObjectInfo::class);
+        foreach ($this->getChildObjectsForSchema($db, $path[0]) as $object) {
+            $objects[] = $object;
+        }
+        $this->assertDatabaseExists($objects, $db, $path[0]);
+        $infoObject->setObjects($objects);
+        $response->setSchemaInfo($infoObject);
+        return $response;
+    }
+
+    /**
+     * @param string[] $path
+     */
+    private function getTableResponse(array $path, ObjectInfoResponse $response, Connection $db): ObjectInfoResponse
+    {
+        assert(count($path) === 2, 'Error path must have exactly two elements.');
+        try {
+            $response->setTableInfo(TableReflectionResponseTransformer::transformTableReflectionToResponse(
+                $path[0],
+                new TeradataTableReflection(
+                    $db,
+                    $path[0],
+                    $path[1]
+                )
+            ));
+        } catch (DBALException $e) {
+            if (strpos($e->getMessage(), 'does not exist.') !== false) {
+                throw new ObjectNotFoundException($path[1]);
+            }
+            throw $e;
+        }
+        return $response;
+    }
+
+    /**
+     * @param string[] $path
+     */
+    private function getViewResponse(array $path, ObjectInfoResponse $response): ObjectInfoResponse
+    {
+        assert(count($path) === 2, 'Error path must have exactly two elements.');
+        $infoObject = new ViewInfo();
+        // todo: set view props
+        $response->setViewInfo($infoObject);
+        return $response;
+    }
+
+    private function assertDatabaseExists(RepeatedField $objects, Connection $db, string $databaseName): void
+    {
+        if ($objects->count() === 0) {
+            // test if database exists
+            $res = $db->fetchOne(sprintf(
+                'SELECT DatabaseName FROM DBC.DatabasesVX WHERE DatabaseName = %s;',
+                TeradataQuote::quote($databaseName)
+            ));
+            if ($res === false) {
```

Comment:
Nestacilo by pouzit \Keboola\StorageDriver\FunctionalTests\BaseCase::isDatabaseExists?

Reply:
Hehe to sem zapomněl, že sem napsal :) Výsledek je stejný tak je to imho jedno, ideální by bylo, kdyby to bylo v utils takové věci.